src/analyze/analyze.c

```diff
@@ -371,25 +376,24 @@ static int acquire_time_data(sd_bus *bus, struct unit_times **out) {
                         NULL);
         if (r < 0) {
                 log_error("Failed to list units: %s", bus_error_message(&error, -r));
-                goto fail;
+                return r;
         }
 
         r = sd_bus_message_enter_container(reply, SD_BUS_TYPE_ARRAY, "(ssssssouso)");
         if (r < 0) {
                 bus_log_parse_error(r);
-                goto fail;
+                return r;
         }
 
         while ((r = bus_parse_unit_info(reply, &u)) > 0) {
                 struct unit_times *t;
 
-                if (!GREEDY_REALLOC(unit_times, size, c+1)) {
-                        r = log_oom();
-                        goto fail;
-                }
+                if (!GREEDY_REALLOC(unit_times, size, c+1))
```

Comment:
... and then this would have to be `GREEDY_REALLOC(unit_times, size, c + 2)`, to keep the last entry empty.

Reply:
Oh of course, I thought `c+1` would make `unit_times[c+1]` a valid buffer, but that's of course not how that works... Thanks!